src/libraries/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -2077,19 +2077,112 @@ public async Task SendHttpGetRequest_WithCachePolicyBypassCache_DoNotAddHeaders(
+                    var stream = await request.GetRequestStreamAsync();
+                },
+                async (server) => 
+                {
+                    Exception? exception = await Record.ExceptionAsync(() => server.AcceptConnectionAsync(_ =>
+                    {
+                        return Task.CompletedTask;
+                    }));
+                    Assert.Null(exception);
+                }
+            );
+        }
+
+        [Fact]
+        public async Task SendHttpPostRequest_AllowWriteBufferingFalse_ConnectionShouldStartWithRequestStream()
+        {
+            const string text = "Hello World!!!!\n";
+            await LoopbackServer.CreateClientAndServerAsync(
+                async (uri) =>
+                {
+                    const int size = 1048576 * 2;
+                    HttpWebRequest request = WebRequest.CreateHttp(uri);
+                    request.ContentType = "application/text";
+                    request.ContentLength = size;
+                    request.Method = "POST";
+                    request.AllowWriteStreamBuffering = false;
+                    using var stream = await request.GetRequestStreamAsync();
+                    for (int i = 0; i < size / 16; i++)
+                    {
+                        await stream.WriteAsync(new ReadOnlyMemory<byte>(Encoding.UTF8.GetBytes(text)));
+                    }
+                },
+                async (server) =>
+                {
+                    await server.AcceptConnectionAsync(async connection =>
+                    {
+                        var data = await connection.ReadRequestDataAsync();
+                        Assert.Equal(text, Encoding.UTF8.GetString(data.Body[0..text.Length]));
+                    });
+                }
+            );
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(0.1)]
+        public async Task SendHttpPost_WriteBufferingEnabled(int delaySec)
+        {
+            const string text = "Hello World!!!!\n";
+            await LoopbackServer.CreateClientAndServerAsync(
+                async (uri) =>
+                {
+                    const int size = 1048576 / 4;
+                    HttpWebRequest request = WebRequest.CreateHttp(uri);
+                    request.AllowWriteStreamBuffering = true;
+                    request.ContentType = "application/text";
+                    request.ContentLength = size;
+                    request.Method = "POST";
+                    request.Timeout = (size/16 * delaySec) + request.Timeout;
+                    using var stream = await request.GetRequestStreamAsync();
+                    for (int i = 0; i < size/16; i++)
+                    {
+                        await stream.WriteAsync(new ReadOnlyMemory<byte>(Encoding.UTF8.GetBytes(text)));
+                        if (delaySec > 0)
+                        {
+                            await Task.Delay(TimeSpan.FromSeconds(delaySec));
```

Comment:
What is the behavior being tested with the delays?

Reply:
Just making sure that, waiting for `StreamContent` on `HttpClient` will not cause any unexpected behavior.